### PR TITLE
Сode optimization from issue #11643

### PR DESCRIPTION
**Pull request by nvyush**
Pull Request for Issue #11643.
### Summary of Changes

1) `array(...)` and `in_array(...)` replaced with more effective in that case bitwise operations.
2) initialization of `$denied_states` move before of the cycle.

Modifications of code are trivial and should not lead to side effects.
### Testing Instructions

1) Create two test users e.g. 'test1' and 'test2'.
2) Set user 'test2' as blocked (set 'Accout Details' - 'Block this User' to 'Yes').
3) Try to login as user 'test1'. You should can do this.
4) Try to login as user 'test2'. You should get message "Login denied! Your account has either been blocked or you have not activated it yet." (JERROR_NOLOGIN_BLOCKED).
### Documentation Changes Required

Not needed.

### CODE REVIEW

**Comment by BurtNL** [comment]
I have tested this item :white_check_mark: successfully on 2e28ca78c52d262ee5d0b1c04cbcc00071bc9c22<br /><br />Installed a clean 3.6.3-dev.
Created user1 and user2, set user2 to blocked.
Logged in on frontend as user1: OK!
Logged in on frontend as user2: message as stated in this PR, access denied.

Applied the patch.
Created user3 and user4, set user4 to blocked.
Logged in on frontend as user3: OK!
Logged in on frontend as user4: message as stated in this PR, access denied.

Conclusion: I did not find any difference, everything works as expected. Tested successfully.
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11655">issues.joomla.org/joomla-cms/11655</a>.</sub>

**Comment by csthomas** [comment]
I have tested this item :white_check_mark: successfully on 2e28ca78c52d262ee5d0b1c04cbcc00071bc9c22<br /><br />I have tested it on J3.6.2 with com_patchtester.
<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11655">issues.joomla.org/joomla-cms/11655</a>.</sub>

**Comment by nvyush** [comment]
I updated branch accidentally without changing any code. Please, recheck.

File: libraries/cms/application/cms.php [hunk 1]
```diff
@@ -817,12 +817,11 @@ public function login($credentials, $options = array())
...
 			 */
 			$authorisations = $authenticate->authorise($response, $options);
+			$denied_states = JAuthentication::STATUS_EXPIRED | JAuthentication::STATUS_DENIED;
 
 			foreach ($authorisations as $authorisation)
 			{
-				$denied_states = array(JAuthentication::STATUS_EXPIRED, JAuthentication::STATUS_DENIED);
-
-				if (in_array($authorisation->status, $denied_states))
+				if ((int) $authorisation->status & $denied_states)
```
Comment (dgt41) [inline libraries/cms/application/cms.php]: `&&` I guess instead of `&`

Comment (csthomas) [inline libraries/cms/application/cms.php]: No, this is bitwise operator between integers, no logical

**Comment by csthomas** [comment]
I have tested this item :white_check_mark: successfully on 1a4203259305215e94aed685a2431c1b52d3e20e<br /><br />The same as instruction say.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11655">issues.joomla.org/joomla-cms/11655</a>.</sub>

**Comment by BhaveshTailored** [comment]
I have tested successfully.
I did not find any difference, everything works as expected.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11655">issues.joomla.org/joomla-cms/11655</a>.</sub>

**Comment by bhavikTailored** [comment]
I have created two user like  "Test5" and "Test6" and set Test6 to blocked.
- I did not find any difference and working is fine.
screen shot 2016-08-24 at 02 13 15<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11655">issues.joomla.org/joomla-cms/11655</a>.</sub>

**Comment by bhavikTailored** [comment]
I have tested this item :white_check_mark: successfully on 1a4203259305215e94aed685a2431c1b52d3e20e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11655">issues.joomla.org/joomla-cms/11655</a>.</sub>

**Comment by BhaveshTailored** [comment]
I have tested this item :white_check_mark: successfully on 1a4203259305215e94aed685a2431c1b52d3e20e<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11655">issues.joomla.org/joomla-cms/11655</a>.</sub>

**Comment by tomartailored** [comment]
I have tested this item :white_check_mark: successfully on 1a4203259305215e94aed685a2431c1b52d3e20e<br /><br />i have tested successfully this item<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11655">issues.joomla.org/joomla-cms/11655</a>.</sub>

**Comment by brianteeman** [comment]
@tomartailored This has already been merged - so either you are not testing with the current staging which you must always do or you didnt really test it but I am sure that you did didnt you<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/11655">issues.joomla.org/joomla-cms/11655</a>.</sub>

**Pull request merged**